File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetImageListUseCase.java
package com.hwonchul.movie.domain.usecase.details;

import com.hwonchul.movie.domain.model.Image;

import java.util.List;

import io.reactivex.rxjava3.core.Flowable;

/**
 * 영화 스틸컷 가져오기 Usecase
 */
public interface GetImageListUseCase {

    Flowable<List<Image>> invoke(int movieId);
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetImageListUseCase.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.model.Image
import kotlinx.coroutines.flow.Flow

/**
 * 영화 스틸컷 가져오기 Usecase
 */
interface GetImageListUseCase {

    operator fun invoke(movieId: Int): Flow<Result<List<Image>>>
}